decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative('nameable')
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```

Comment:
- [ OPTIONAL ] and here also to call the method for correct name from the interface namable so it could be like this:


> `@nameable.correct_name`